src/macro.jl

```diff
@@ -0,0 +1,117 @@
+function _kwarg_to_tuple(ex)
+    ex.head != :(=) &&
+        throw(ArgumentError("Invalid keyword argument. Doesn't contain '='."))
+    name, val = ex.args
+    !(name isa Symbol) &&
+        throw(ArgumentError("First part of keyword argument isn't a symbol."))
+    val isa QuoteNode && (val = val.value)
+    (name, val)
+end
+
+function _tasklocal_assign_to_exprs(ex)
+    if ex.head != :(=)
+        throw(ErrorException("Wrong usage of @tasklocal. Expected assignment, e.g. `A::Matrix{Float} = rand(2,2)`."))
+    end
+    left_ex = ex.args[1]
+    if left_ex isa Symbol || left_ex.head != :(::)
+        throw(ErrorException("Wrong usage of @tasklocal. Expected typed assignment, e.g. `A::Matrix{Float} = rand(2,2)`."))
+    end
+    tls_sym = left_ex.args[1]
+    tls_type = left_ex.args[2]
+    tls_def = ex.args[2]
+    tls_storage = gensym()
```

Comment:
```suggestion
    @gensym tls_storage
```
this'll do `tls_storage = gensym(:tls_storage)` which'll generate (slightly) more readable code, while still being safe.